test/e2e/replica_set_change_version/replica_set_test.go

```diff
@@ -18,7 +19,15 @@ func TestMain(m *testing.M) {
 
 func TestReplicaSetUpgradeVersion(t *testing.T) {
 	ctx := f.NewContext(t)
-	defer ctx.Cleanup()
+
+	// When the tests are ran in the evg host we don't want to clean up
+	// the context, as we need it to dump diagnostics
+	// In all other cases (e.g. tests ran locally) we still clean up
+	// the context when we finish
+	if os.Getenv("SKIP_CLEANUP") != "1" {
```

Comment:
I think I know we can't escape this env variable for now, but it worries me that we use a env variable to notify the need of a cleanup.